src/triton_cli/parser.py

```diff
@@ -496,8 +465,97 @@ def parse_args(argv=None):
     parse_args_repo(subcommands)
     parse_args_server(subcommands)
     parse_args_inference(subcommands)
+    parse_args_optimize(subcommands)
     parse_args_profile(subcommands)
     parse_args_utils(subcommands)
     add_verbose_args([parser])
-    args = parser.parse_args(argv)
+
+    argv_ = argv if argv is not None else sys.argv[1:]
+    # If a passthrough command is passed as the first arg,
+    # special handling is needed.
+    need_special_handling = len(argv_) > 1 and argv_[0] in PASSTHROUGH_SUBCOMMANDS
+    if need_special_handling:
+        updated_args, pruned_args = prune_extra_subcommand_args(
+            argv_, subcommands.choices.keys()
+        )
+        args, unknown_args = parser.parse_known_args(updated_args)
+        args = add_unknown_args_to_args(args, pruned_args, unknown_args)
+    else:
+        args = parser.parse_args(argv)
```

Comment:
This one should be fine, because if argv is None, it will read from the command line. This preserves the regular behavior for all commands that do not require pass-through behavior.